### PR TITLE
[fix](compaction) tablet is not removed in compaction queue if dropped

### PR DESCRIPTION
**Pull request by mymeiyi**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
it will cause the data dir can not be deleted because the table shared ptr use count is 2

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by zhannngchen — APPROVED** [review]
LGTM

File: be/src/olap/olap_server.cpp [hunk 1]
```diff
@@ -1071,7 +1071,8 @@ Status StorageEngine::_submit_compaction_task(TabletSharedPtr tablet,
             if (!tablet->can_do_compaction(tablet->data_dir()->path_hash(), compaction_type)) {
                 LOG(INFO) << "Tablet state has been changed, no need to begin this compaction "
                              "task, tablet_id="
-                          << tablet->tablet_id() << "tablet_state=" << tablet->tablet_state();
+                          << tablet->tablet_id() << ", tablet_state=" << tablet->tablet_state();
+                _pop_tablet_from_submitted_compaction(tablet, compaction_type);
```
Comment (TangSiyang2001) [inline be/src/olap/olap_server.cpp:1075]: Could we use a Defer to call `_pop_tablet_from_submitted_compaction`

**Pull request merged**